improver/utilities/spatial.py

```diff
@@ -169,13 +169,28 @@ class DifferenceBetweenAdjacentGridSquares(BasePlugin):
     individually.
     """
 
-    def __init__(self, gradient=False):
-        """
-        Initialise class.
+    @staticmethod
+    def _update_metadata(diff_cube, coord_name, cube_name):
+        """Rename cube, add attribute and cell method to describe difference.
+
+        Args:
+            diff_cube (iris.cube.Cube)
+            coord_name (str)
+            cube_name (str)
+
         """
-        self.is_gradient = gradient
+        # Add metadata to indicate that a difference has been calculated.
+        # TODO: update metadata for difference when
+        #  proper conventions have been agreed upon.
+        cell_method = CellMethod(
+            "difference", coords=[coord_name], intervals="1 grid length"
+        )
+        diff_cube.add_cell_method(cell_method)
+        diff_cube.attributes["form_of_difference"] = "forward_difference"
+        diff_cube.rename("difference_of_" + cube_name)
 
-    def create_difference_cube(self, cube, coord_name, diff_along_axis):
+    @staticmethod
+    def create_difference_cube(cube, coord_name, diff_along_axis):
```

Comment:
I can't find anything testing that the modified coord points are as expected (half-way between the original points). While not added by this PR, the method is now being used more widely so I'd like to see such a test included please.

Reply:
I've added a check that the modified coordinate points are half-way between the original points, as suggested.